invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/RDMDepositForm.js

```diff
@@ -311,18 +267,88 @@ export class RDMDepositForm extends Component {
                 <VersionField />
                 <PublisherField />
               </AccordionField>
-              {/**TODO: uncomment to use FundingField*/}
-              {/* <AccordionField
-              active={true}
-              label={"Funding"}
+
+              <AccordionField
+                fieldPath=""
+                active={true}
+                label={"Funding"}
+                ui={this.accordionStyle}
               >
-              <FundingField options={this.vocabularies.metadata.funding} />
-              <ComingSoonField
-                fieldPath="metadata.funding"
-                label="Awards"
-                labelIcon="money bill alternate outline"
-              />
-              </AccordionField> */}
+                <FundingField
+                  fieldPath="metadata.funding"
+                  searchConfig={{
+                    searchApi: {
+                      axios: {
+                        headers: {
+                          //  FIXME use for internationalisation
+                          //  Accept: "application/vnd.inveniordm.v1+json"
+                          Accept: "application/json",
+                        },
+                        url: "/api/awards",
+                        withCredentials: false,
+                      },
+                    },
+                    initialQueryState: {
+                      sortBy: "bestmatch",
+                      sortOrder: "asc",
+                      layout: "list",
+                      page: 1,
+                      size: 5,
+                    },
+                  }}
+                  label="Awards"
+                  labelIcon="money bill alternate outline"
+                  deserializeAward={(award) => {
+                    return {
+                      title: award.title.en ?? award.title,
+                      pid: award.pid,
+                      number: award.number,
+                      funder: award.funder ?? "",
+                      id: award.id,
+                      ...(award.identifiers && {
+                        identifiers: award.identifiers,
+                      }),
+                      ...(award.acronym && { acronym: award.acronym }),
+                    };
+                  }}
+                  deserializeFunder={(funder) => {
+                    return {
+                      id: funder.id,
+                      name: funder.name,
+                      ...(funder.pid && { pid: funder.pid }),
+                      ...(funder.country && { country: funder.country }),
+                      ...(funder.identifiers && {
+                        identifiers: funder.identifiers,
+                      }),
+                    };
+                  }}
+                  computeFundingContents={(funding) => {
```

Comment:
I find this function quite hard to read, would it be possible to simplify ?

Reply:
Trying... but unsure, the logic is pretty much specific

Reply:
re-worked, I think is more clear now :)